### PR TITLE
Add chain for anchors

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Add clickable chain icon for anchors in docs and blogs.
- Fix broken links in `0.5` and `0.4` releases.

**Related issue(s)**
See kyma-project/website#103
See #139

### CODE REVIEW

File: src/api-extensions/fix-versions.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
+const linksToFixing = {
```
Comment (pkosiec) [inline src/api-extensions/fix-versions.js]: Rename to `linksToRewrite`

File: src/api-extensions/links-parser.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        doc.source,
+        version,
+        contentType,
+        id,
+        doc.type,
+        doc.title,
+      );
+    }
+
+    doc.source = replaceRelativeHrefForAssetsToAbsolute(
```
Comment (pkosiec) [inline src/api-extensions/links-parser.js]: What do you think about `makeAbsolutePathsForAssets`?

File: src/api-extensions/fix-versions.js (hunk 1 above)
Comment (pkosiec) [inline src/api-extensions/fix-versions.js]: `fix-versions` -> `links-rewrite` or something like that would be a better name

File: src/api-extensions/links-parser.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+    doc.source = changeVersionInLinksHref(
+      doc.source,
+      version,
+      includeVersionInPath,
+    );
+    doc.source = changeHeadersID(doc.source, doc.type, doc.title);
+    doc.source = replaceHeadersToHeadersWithChains(doc.source);
+
+    const versionToFixing = ["0.4", "0.5"];
```
Comment (pkosiec) [inline src/api-extensions/links-parser.js]: I would just go through the `links-rewrite` file and see if there is a rewrite rule for a specific version. Then the array is unnecessary

File: src/api-extensions/links-parser.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    if (versionToFixing.find(el => el === version)) {
+      doc.source = fixVersion(
```
Comment (pkosiec) [inline src/api-extensions/links-parser.js]: please rename `fixVersion` to `rewriteLinks`

File: src/api-extensions/links-parser.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+function changeHeadersID(source, docType, docTitle) {
+  if (!docType) docType = docTitle;
+
+  source = source.replace(headerIDRegexp, occurrence => {
+    headerIDRegexp.lastIndex = 0;
+    let id = headerIDRegexp.exec(occurrence);
+
+    if (!id || !id[2]) return occurrence;
+    id = id[2];
```
Comment (pkosiec) [inline src/api-extensions/links-parser.js]: I don't get the logic here :smile:

Comment (magicmatatjahu) [reply src/api-extensions/links-parser.js]: `headerIDRegexp` has a value `/id=(\"|')(.*?)(\"|')/g`. `Regexp.exec()` function performs search operations for a given string. The result is a array. In this case we have 3 options: [(\"|'), (.*?), (\"|')]. The first item of array is always whole matched string, eg:
`id="example-id"` -> `[id="example-id", '"', "example-id",  '"']`

Comment (pkosiec) [reply src/api-extensions/links-parser.js]: Thanks, now it makes sense 😄

**Comment by pkosiec** [comment]
@michal-hudy and I were talking about the link rewriting. Unfortunately, it shouldn't be done on website, as they will be broken on Docs UI. So the only solution would be fixing those links on release branches and doing a release.

Anyway, please ask @derberg if he is okay with the current approach, as in new 0.6 release the links will be OK as you fixed them on master. If he agrees, then I will approve this PR 🙂

**Pull request merged**